### PR TITLE
Replace EnginesBuilder with EnginesConfigBuilder.

### PR DESCRIPTION
**Pull request by fhwang**
Giving builder more flexibility in how Engine instances are instantiated will make it easier to make certain classes more cohesive. This refactor will be followed up by one on the builder side.

Note that this PR breaks backwards compatibility, so I'll follow it up soon with a small refactor in builder.

### CODE REVIEW

**Comment by fhwang** [comment]
@codeclimate/review this is ready for a look, thanks!

File: lib/cc/analyzer/engines_config_builder.rb [hunk 1]
```diff
@@ -11,9 +19,11 @@ def initialize(registry:, config:, container_label:, source_dir:, requested_path
         @source_dir = source_dir
       end
 
-      def run(engine_class = Analyzer::Engine)
+      def run
         names_and_raw_engine_configs.map do |name, raw_engine_config|
-          build_engine(engine_class, name, raw_engine_config)
+          label = @container_label || SecureRandom.uuid
```
Comment (GordonDiggs) [inline lib/cc/analyzer/engines_config_builder.rb]: What do you think of doing the `|| SecureRandom.uuid` in the constructor?

Comment (fhwang) [inline lib/cc/analyzer/engines_config_builder.rb]: The desired behavior, based at least on the previous code, was to call `SecureRandom.uuid` once for each engine, so I'm just preserving that here.

Comment (GordonDiggs) [inline lib/cc/analyzer/engines_config_builder.rb]: Ah got it

**Comment by GordonDiggs** [comment]
LGTM

**Pull request merged**